lib/dpl/provider/now.rb

```diff
@@ -0,0 +1,136 @@
+module DPL
+    class Provider
+      class Now < Provider
+        npm_g 'now'
+  
+        def directory
+          File.expand_path( (context.env['TRAVIS_BUILD_DIR'] || '.' ) + '/' + (options[:directory] || '') )
+        end
+  
+        def deploy_name
+          options[:deploy_name]
+        end
+  
+        def team
+          if options[:team]
+            log "> Adding custom team scope #{options[:team]}"
+            now_team_option = "--team #{options[:team]}"
+          else
+            now_team_option = ''
+            log '> No custom team scope provided.'
+          end
+
+          return now_team_option
+        end
+  
+        def type
+          options[:type]
+        end
+  
+        def alias_url
+          options[:alias]
+        end
+  
+        def cleanup
+          options[:cleanup]
+        end
+  
+        def scale
+          options[:scale]
+        end
+
+        def rules_domain
+          options[:rules_domain]
+        end
+
+        def rules_file
+          options[:rules_file]
+        end
+  
+        def token
+          context.env['NOW_TOKEN']
+        end
+
+        def auth
+          return "--token #{token} #{team}"
```

Comment:
This is a bit strange. The invocation of `#team` here interlaces the log messages defined in that method. As `auth` is called from a few spots, I expect it to be quite noisy.

The credentials should be cached and logged at most once.